pyaedt/modules/Material.py

```diff
@@ -1992,16 +1993,248 @@ def set_magnetic_coercivity(self, value=0, x=1, y=0, z=0):
         return self.update()
 
     @pyaedt_function_handler()
-    def set_electrical_steel_coreloss(self, kh=0, kc=0, ke=0, kdc=0, cut_depth=0.0001):
-        """Set Electrical Steel Type Core Loss.
+    def get_core_loss_coefficients(
+        self,
+        points_list_at_freq,
+        core_loss_model_type="Electrical Steel",
+        thickness="0.5mm",
+        conductivity=0,
+        coefficient_setup="w_per_cubic_meter",
+    ):
+        """Compute core loss coefficients from the loss characteristics (B-P Curve) at a given frequency.
+
+        Parameters
+        ----------
+        points_list_at_freq : dict
+            points_list_at_freq is a dictionary of BP values for a given frequency.
+            Key is the frequency and values are the list of points (BP curve).
+        core_loss_model_type : str, optional
+            Core loss model type. It can be ``Electrical Steel`` or ``Power Ferrite``.
+            The default value is ``Electrical Steel``.
+        thickness : str, optional
+            Thickness provided as value + unit.
+            The default value is ``0.5mm``.
+        conductivity : float, optional
+            Material conductivity.
+            The default value is ``0``.
+        coefficient_setup : str, optional
+            Core loss unit.
+            Possible values are: ``w_per_cubic_meter``, ``kw_per_cubic_meter``, ``w_per_lb``, ``w_per_kg``.
+            The default value is ``w_per_cubic_meter``.
```

Comment:
```suggestion
        points_list_at_freq : dict
            Dictionary of BP values for a given frequency. The key is the
            frequency, and the values are the list of points (BP curve).
        core_loss_model_type : str, optional
            Core loss model type. The default value is ``Electrical Steel``.
            Options are ``Electrical Steel`` and ``Power Ferrite``.
        thickness : str, optional
            Thickness provided as the value plus the unit.
            The default is ``0.5mm``.
        conductivity : float, optional
            Material conductivity.
            The default is ``0``.
        coefficient_setup : str, optional
            Core loss unit. The default is ``"w_per_cubic_meter"``.
            Options are ``"kw_per_cubic_meter"``, ``"w_per_cubic_meter"``, ``"w_per_kg"``,
            and ``"w_per_lb"``.

```
List the default first. Then list all options in alphabetical order.